### DISCUSSION TITLE
Is there an elegant way to redefine colon : as division ?

### DISCUSSION THREAD

**Discussion by nickkolok** [josdejong/mathjs#3023] [Q&A]
I need to parse school-ish expressions like `6:3` (that's 2). I know that `:` is used in mathjs for ranges.

I can't just `.replace(/:/g, '/')` because I need to apply some simplification rules to the parsed expression and then render the expression as $\TeX$; this replacement results in `\frac`s in the output.

**Comment by josdejong** [comment]
What you basically can do is:

```js
const expression = '6:3'
const node = math.parse(expression)
const transformedNode = node.transform(node => {
  // add your logic here to check for RangeNode and replace it with an OperatorNode doing a division when applicable
})
const result = transformedNode.evaluate()
```

You can find the documation about `.transform(callback)` here: https://mathjs.org/docs/expressions/expression_trees.html
